blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -138,18 +135,13 @@ class RunDetails extends Component {
     };
 
     afterClose = () => {
-        const { router, location, params } = this.context;
-        const fallbackUrl = buildPipelineUrl(params.organization, params.pipeline);
-
-        location.pathname = this.opener || fallbackUrl;
-
-        // reset query
-        /*
-         FIXME: reset query when you go back, we may want to store the whole location object in previous so we have a perfect prev.
-         this.opener would then be location and we the above location = this.opener || { pathname: fallbackUrl]
-         */
-        location.query = null;
-        router.push(location);
+        const { router, params } = this.context;
+        if (this.opener) {
+            router.goBack();
+        } else {
+            const fallbackUrl = buildPipelineUrl(params.organization, params.pipeline);
+            router.push(fallbackUrl);
```

Comment:
hmm, I'm not really sure, either. replace might be more consistent, but if i go to the page directly, and accidentally hit esc or something, it would be nice to go back in that case.